jobs/worker.go

```diff
@@ -91,3 +98,45 @@ func RunJobNow(j Job) {
 	}
 	l.Log.Infof("Finished Job %v with %v", j.Name(), j.Arguments())
 }
+
+// healthCheckStruct is a struct that contains the time for the health checker and a channel to pass the shared memory for the time through time channel
+type healthCheckStruct struct {
+	timeStamp   time.Time
+	timeChannel chan time.Time
+}
+
+func (hc *healthCheckStruct) healthChecker() {
+	//run healthcheck loop and consumer in the background
+	go hc.healthcheckLoop()
+	go hc.healthCheckConsumer()
+
+	//create new GET request through echo with the same path name "/health"
+	e := echo.New()
+	e.GET("/health", func(c echo.Context) error {
+		//health should return a 500 if there hasn't been a response from ping request within 30 seconds
+		if hc.timeStamp.Before(time.Now().Add(-30 * time.Second)) {
+			return c.String(http.StatusInternalServerError, "500")
+		}
+		return c.String(http.StatusOK, "OK")
+
+	})
+	l.Log.Fatal(e.Start(":8000"))
+}
+
+// healthcheckConsumer runs a loop against the healthcheck time channel and updates the shared timestamp with the last time we received a response
+func (hc *healthCheckStruct) healthCheckConsumer() {
+	for {
```

Comment:
We can probably change this to `for range hc.timeChannel` since we don't have to manually pull from the channel anymore.